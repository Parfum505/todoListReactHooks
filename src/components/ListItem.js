import React from "react";

const ListItem = ({ task, setCompleted, deleteTask }) => (
  <li className="input-group mb-3" id={task.id}>
    <div className="custom-control custom-checkbox">
      <input
        type="checkbox"
        id={`customCheck_${task.id}`}
        className="custom-control-input"
        checked={task.completed}
        onChange={() => setCompleted(task.id)}
      />
      <label
        className="custom-control-label"
        htmlFor={`customCheck_${task.id}`}
      >
        &nbsp;
      </label>
    </div>
    <input
      type="text"
      className={`form-control ${task.completed ? "completed" : ""}`}
      value={task.item}
      readOnly
    />
    <div className="input-group-append">
      <button
        className="btn btn-danger input-group-text"
        onClick={() => deleteTask(task.id)}
      >
        X
      </button>
    </div>
  </li>
);

export default React.memo(ListItem);
