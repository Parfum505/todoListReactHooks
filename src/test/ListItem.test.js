import React from "react";
import ListItem from "../components/ListItem";
import { render, cleanup, fireEvent } from "@testing-library/react";

afterEach(cleanup);

describe("ListItem testing", () => {
  test("matches snapshot", () => {
    const fakeTask = { id: 1, item: "to learn Jest", completed: false };
    const { asFragment } = render(<ListItem task={fakeTask} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders listItem", () => {
    const fakeTask = { id: 1, item: "to learn Jest", completed: false };
    const { queryByTestId, queryByDisplayValue } = render(
      <ListItem task={fakeTask} />
    );
    expect(queryByDisplayValue("to learn Jest")).toBeInTheDocument();
    expect(queryByTestId("listItem")).toBeInTheDocument();
  });
  test("checkbox, deleteButton clicked", () => {
    const fakeTask = { id: 1, item: "to learn Jest", completed: false };
    const mockSetCompleted = jest.fn();
    const mockDeleteTask = jest.fn();
    const { queryByTestId } = render(
      <ListItem
        task={fakeTask}
        setCompleted={mockSetCompleted}
        deleteTask={mockDeleteTask}
      />
    );
    fireEvent.click(queryByTestId("checkbox"));
    expect(mockSetCompleted).toBeCalled();
    fireEvent.click(queryByTestId("deleteBtn"));
    expect(mockDeleteTask).toBeCalled();
  });
});
