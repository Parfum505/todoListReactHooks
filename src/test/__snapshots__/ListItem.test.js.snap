// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem testing matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="input-group mb-3"
    data-testid="listItem"
    id="1"
  >
    <div
      class="custom-control custom-checkbox"
    >
      <input
        class="custom-control-input"
        data-testid="checkbox"
        id="customCheck_1"
        type="checkbox"
      />
      <label
        class="custom-control-label"
        for="customCheck_1"
      >
        Â 
      </label>
    </div>
    <input
      class="form-control "
      data-testid="taskInput"
      readonly=""
      type="text"
      value="to learn Jest"
    />
    <div
      class="input-group-append"
    >
      <button
        class="btn btn-danger input-group-text"
        data-testid="deleteBtn"
      >
        X
      </button>
    </div>
  </li>
</DocumentFragment>
`;
